### ISSUE TITLE
auto_route_generator >=1.0.0 is incompatible with json_serializable 4.0.3

### ISSUE THREAD

**Issue by pizzaldi** [Milad-Akarie/auto_route_library#410]
> Because auto_route_generator >=1.0.0 depends on build ^1.5.0 and json_serializable 4.0.3 depends on build ^2.0.0, auto_route_generator >=1.0.0 is incompatible with json_serializable 4.0.3.

I'm using both auto_route_generator and json_serializable in my project, but the latest version of auto_route_generator did not use the same version of build as json_serializable. Is it possible to update this?

Thenk you in advance.

**Comment by Milad-Akarie** [comment]
@pizzaldi I initially kept it ^1.5.0 so other generator packages won't face the same issue we're facing now lol 
I'll pomp it up in the next build, for now you could just override the deps.

```yaml
dependency_overrides:
  build: ^2.0.0
```

**Comment by zs-dima** [comment]
The same issue with json_serializable 4.1.0
dependency_overrides quite unsafe so it looks like temporary solution

```
Because no versions of auto_route_generator match >1.0.1 <2.0.0 and auto_route_generator 1.0.1 depends on source_gen ^0.9.10+3, auto_route_generator ^1.0.1 requires source_gen ^0.9.10+3.
And because json_serializable >=4.1.0 depends on source_gen ^1.0.0, auto_route_generator ^1.0.1 is incompatible with json_serializable >=4.1.0.
So, because workshop depends on both json_serializable ^4.1.0 and auto_route_generator ^1.0.1, version solving failed.
pub get failed (1; So, because workshop depends on both json_serializable ^4.1.0 and auto_route_generator ^1.0.1, version solving failed.)
```

**Comment by Milad-Akarie** [comment]
@zs-dima  I'm just working on some fixes to be included in the next build. this one has already been handled.

**Comment by Milad-Akarie** [comment]
Should be fixed in auto_route: 1.0.2
Thanks for reporting

**Comment by zs-dima** [comment]
@Milad-Akarie thank a lot for the fast fix
